mm2src/coins/z_coin.rs

```diff
@@ -678,7 +715,7 @@ impl ZCoin {
         else {
             return Ok(false);
         };
-        if &address == expected_address {
+        if &address != expected_address {
```

Comment:
When was this changed? This is a bad logical bug that we had that caused swaps failing I assume. I see it was added here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2112 I guess we never caught it in testing for the previous release since pirate swaps were not working c.c. @dimxy

Reply:
Oh, it's my code (bad refactoring): 2873ac32ac738d392cb7155a7cb366fd0d054f38.
I should add a test for this case

Reply:
I actually found the bug while testing z_coin docker tests.